pkg/util/chunkinfologger/debugheader_test.go

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package chunkinfologger
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/grafana/dskit/middleware"
+	"github.com/stretchr/testify/require"
+)
+
+func TestChunkInfoLoggerMiddleware(t *testing.T) {
+	testCases := map[string]struct {
+		headers       http.Header
+		expectedOk    bool
```

Comment:
[nit] `expectedLoggingEnabled` might be clearer

(similar thing applies to `recordedOk` below)

Reply:
suggest: expectedEnabled, expectedLabels and actualEnabled, actualLabels for short

Reply:
Sounds good to me.